modules/data-streams/src/yamlRestTest/resources/rest-api-spec/test/data_stream/200_rollover_failure_store.yml

```diff
@@ -323,8 +371,33 @@ teardown:
   - match: { data_streams.0.name: data-stream-for-rollover }
   - match: { data_streams.0.timestamp_field.name: '@timestamp' }
   # Both backing and failure indices use the same generation field.
-  - match: { data_streams.0.generation: 1 }
+  - match: { data_streams.0.generation: 2 }
+  - length: { data_streams.0.indices: 1 }
+  - match: { data_streams.0.indices.0.index_name: '/\.ds-data-stream-for-rollover-(\d{4}\.\d{2}\.\d{2}-)?000001/' }
+  - length: { data_streams.0.failure_store.indices: 1 }
+  - match: { data_streams.0.failure_store.indices.0.index_name: '/\.fs-data-stream-for-rollover-(\d{4}\.\d{2}\.\d{2}-)?000002/' }
+
+---
+"Rolling over an uninitialized failure store should initialize it":
```

Comment:
Can we add one more test case here for if we try to roll over a data stream's failure store and the data stream does not have failure store enabled? I imagine that it should work the same as a data stream with the feature enabled but uninitialized. 

I recall we previously discussed treating the failure store enabled flag on the data stream as a guard for redirecting failures to the failure store, but I think all the other facilities should continue to work regardless of that flag's value.

Reply:
Good suggestion, thanks!